### DISCUSSION TITLE
create seedvault-compatible backup from non-seedvault rooted device

### DISCUSSION THREAD

**Discussion by retracile** [seedvault-app/seedvault#270] [Q&A]
I would like to create a seedvault-compatible backup of the data on a device that does not have seedvault (such as one running LineageOS 17), but on which I _do_ have root.  And then restore it on a rooted device with seedvault (such as LineageOS 18).

Before I sink too much time into this, I was wondering what tools might already exist?

I see there's some python code for parsing the backups (https://github.com/tlambertz/seedvault_backup_parser).  Are there similar tools for creating backups?  Is the file format documented in detail somewhere?

**Comment by grote** [comment]
If you have root, you might be able to install Seedvault for the restore. 

Possibly related: https://github.com/seedvault-app/seedvault/issues/9

**Comment by Micha-Btz** [comment]
If you have root, than you can checkout https://github.com/machiav3lli/oandbackupx/
No need to create another app.
I use both, since seedvault can't backup apps that prevent it.

**Reply by retracile** [reply]
Thank you; I wasn't aware of that app.  Interesting that you use both... does oandbackupx miss something that seedvault has?
